example/spec/testAPI/APISuccessSpec.js

```diff
@@ -0,0 +1,144 @@
+const { api: apiTestUtils } = require('@cumulus/integration-tests');
+
+/**
+ * Set up files in the S3 data location for a new granule to use for this
+ * test. Use the input payload to determine which files are needed and
+ * return updated input with the new granule id.
+ *
+ * @param {string} bucket - data bucket
+ * @param {string} granuleId - granule id for the new files
+ * @param {string} inputPayloadJson - input payload as a JSON string
+ * @returns {Promise<Object>} - input payload as a JS object with the updated granule ids
+ */
+async function setupTestGranuleForAPI(bucket, granuleId, inputPayloadJson) {
+  const baseInputPayload = JSON.parse(inputPayloadJson);
+
+  await createGranuleFiles(
+    baseInputPayload.granules[0].files,
+    bucket,
+    testDataGranuleId,
+    granuleId
+  );
+
+  const updatedInputPayloadJson = globalReplace(inputPayloadJson, testDataGranuleId, granuleId);
+
+  return JSON.parse(updatedInputPayloadJson);
+}
+
+async function waitForExist(CMRLink, outcome, retries) {
+  if (retries === 0) {
+    console.log('Out of retries.');
+    return false;
+  }
+  const existsCheck = await conceptExists(CMRLink);
+  console.log(existsCheck, ' ', outcome);
+  if (existsCheck !== outcome) {
+    sleep.sleep(2);
+    console.log('Retrying...');
+    return waitForExist(CMRLink, outcome, (retries - 1));
+  }
+  else {
+    return true;
+  }
+}
+
+describe('The Cumulus API', () => {
+  let workflowExecution = null;
+  const collection = { name: 'MOD09GQ', version: '006' };
+  const provider = { id: 's3_provider' };
+  const inputPayloadFilename = './spec/testAPI/testAPI.input.payload.json';
+  let inputPayload;
+  let granuleId;
+
+  beforeAll(async () => {
+    console.log('Starting API test');
+    granuleId = randomStringFromRegex(granuleRegex);
+
+    console.log(`granule id: ${granuleId}`);
+
+    const inputPayloadJson = fs.readFileSync(inputPayloadFilename, 'utf8');
+    inputPayload = await setupTestGranuleForAPI(config.bucket, granuleId, inputPayloadJson);
+
+    workflowExecution = await buildAndExecuteWorkflow(
+      config.stackName, config.bucket, taskName, collection, provider, inputPayload
+    );
+  });
+
+  afterAll(async () => {
+    // Remove the granule files added for the test
+    await Promise.all(
+      inputPayload.granules[0].files.map((file) =>
+        s3().deleteObject({
+          Bucket: config.bucket, Key: `${file.path}/${file.name}`
+        }).promise())
+    );
+  });
+  it('completes execution with success status', () => {
+    expect(workflowExecution.status).toEqual('SUCCEEDED');
+  });
+
+  it('makes the granule available through the Cumulus API', async () => {
+    const granule = await apiTestUtils.getGranule({
+      prefix: config.stackName,
+      granuleId: inputPayload.granules[0].granuleId
+    });
+
+    expect(granule.granuleId).toEqual(inputPayload.granules[0].granuleId);
+  });
+
+  describe('reingest a granule', () => {
```

Comment:
`reingest` and `applyWorkflow` are different actions, the `applyWorkflow` is for ingest-in-place which is not tested in this PR.